haystack/components/routers/conditional_router.py

```diff
@@ -146,15 +157,22 @@ def to_dict(self) -> Dict[str, Any]:
             # output_type needs to be serialized to a string
             route["output_type"] = serialize_type(route["output_type"])
 
-        return default_to_dict(self, routes=self.routes)
+        # set all filter callables to None for serialization
+        custom_filters = {filter_name: None for filter_name in self.custom_filters.keys()}
+
+        return default_to_dict(self, routes=self.routes, custom_filters=custom_filters)
 
     @classmethod
-    def from_dict(cls, data: Dict[str, Any]) -> "ConditionalRouter":
+    def from_dict(
+        cls, data: Dict[str, Any], custom_filters: Optional[Dict[str, Callable]] = None
+    ) -> "ConditionalRouter":
```

Comment:
We can't modify the signature of the `from_dict` method.

Reply:
Reverted.